### ISSUE TITLE
Google Drive syncing just hangs

### ISSUE THREAD

**Issue by jasonlally** [benweet/stackedit#826]
It appears that when creating a Stackedit doc from Google drive, the document is created, but the syncing just hangs. I've had a doc just sitting with the sync indicator running and no successful sync.

**Comment by fghhfg** [comment]
+1

**Comment by wesort** [comment]
+1

**Comment by benweet** [comment]
I can't reproduce the issue. Any timeout? or errors in the javascript console?

**Comment by wesort** [comment]
When I open from Google Driver, here is what's going on in Console:

![stackedit-gdrive-console](https://cloud.githubusercontent.com/assets/3920126/8281318/87ead958-18de-11e5-97d2-f44b5cd65280.png)

```
Document was loaded from Application Cache with manifest https://stackedit.io/cache.manifest
Application Cache Checking event
Application Cache NoUpdate event
'Range.detach' is now a no-op, as per DOM (http://dom.spec.whatwg.org/#dom-range-detach).
Uncaught TypeError: Cannot read property 'substring' of undefined
Failed to load resource: the server responded with a status of 401 (Unauthorized)
Objecterror: "unauthorized"reason: "user_not_authenticated"status: 401__proto__: Object
Google login url is detected: https://accounts.google.com/o/oauth2/postmessageRelay?parent=https%3A%2F%2Fstackedit.io#rpctoken=303636359&forcesecure=1
Second factor url is not detected: https://accounts.google.com/o/oauth2/postmessageRelay?parent=https%3A%2F%2Fstackedit.io#rpctoken=303636359&forcesecure=1
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check http://xhr.spec.whatwg.org/.
Google login url is detected: https://accounts.google.com/o/oauth2/auth?client_id=241271498917-t4t7d07qis…ps%3A%2F%2Fstackedit.io&response_type=token&state=868306387%7C0.3008202145
Second factor url is not detected: https://accounts.google.com/o/oauth2/auth?client_id=241271498917-t4t7d07qis…ps%3A%2F%2Fstackedit.io&response_type=token&state=868306387%7C0.3008202145
```

FYI:
- I'm using a ChromeOS machine (stable version, up-to-date)
- I'm a Google Apps user with 2-step authentication (fully logged in)

**Comment by InQuize** [comment]
open/save to google doesn't work in Chrome, but works in IE. Can't auth. Dropbox is fine.
```
Error: A timeout occurred.
    at o (https://stackedit.io/res-min/main.js:25449:71)
    at https://stackedit.io/res-min/main.js:21351:8
    at Array.forEach (native)
    at Function.C.each.C.forEach (https://stackedit.io/res-min/main.js:3995:46)
    at Object.onPeriodicRun (https://stackedit.io/res-min/main.js:21349:6)
    at https://stackedit.io/res-min/main.js:25117:79(anonymous function) @ main.js:21353C.each.C.forEach @ main.js:3995(anonymous function) @ main.js:21349(anonymous function) @ main.js:25117
```

**Comment by InQuize** [comment]
nvm, it was [Disconnect](https://disconnect.me)'s fault.

**Comment by appdesc** [comment]
OK I had disconnect disabled and still hanging so I deleted it and rebooted - still hangs 
I went through all the extensions deleted a load - still hangs 

chromebook up to date - only logged into one gmail account and connecting to it

anyone got a solution?

**Comment by joshfttb** [comment]
Having the same problem

**Comment by joshfttb** [comment]
![image](https://cloud.githubusercontent.com/assets/2595454/12225297/d8998778-b7d5-11e5-9fba-94d88c4798f7.png)

**Comment by joshfttb** [comment]
<img width="1440" alt="screen shot 2016-01-10 at 20 07 34" src="https://cloud.githubusercontent.com/assets/2595454/12225298/ec036be4-b7d5-11e5-9b6a-e87bde61b194.png">

**Comment by martyzigman** [comment]
Here is what I did to solve it.  I stumbled on it.  I too had the authorization errors:

1.  I set the Auto Sync settings to work on New documents to the same folder I wanted to save files by default.  I don't this mattered though.

2.  On just one doc, I was able to it to sync after resetting the account and performing a basic folder setup.  Then it would break.  However, I would then go to Google Drive, find the one file, and Open With StackEdit in a new window.  Somehow, the auth started working and the syncing appeared fine thereafter.

Note, this has now happened to me both on Win7 and OSX both in chrome.
